src/D2L.Security.OAuth2/Keys/Default/MonoRsaSecurityKey.cs

```diff
@@ -24,15 +28,14 @@ internal sealed class MonoRsaSecurityKey : AsymmetricSecurityKey {
 		}
 
 		public override bool IsAsymmetricAlgorithm( string algorithm ) {
-			switch (algorithm) {
-			case SecurityAlgorithms.DsaSha1Signature:
-			case SecurityAlgorithms.RsaSha1Signature:
-			case SecurityAlgorithms.RsaSha256Signature:
-			case SecurityAlgorithms.RsaOaepKeyWrap:
-			case SecurityAlgorithms.RsaV15KeyWrap:
-				return true;
-			default:
-				return false;
+			switch( algorithm ) {
+				case SecurityAlgorithms.RsaSha1Signature:

```

Comment:
But it also shouldn't speak for all of the other algorithms that it has nothing to do with. This method shouldn't be called unless it's a supported algorithm in the first place.

This is exactly what we did in EcdsaSecurityKey with great success.